Demo/Ser_Des/ser_des.h

```diff
@@ -0,0 +1,104 @@
+#ifndef SER_DES_H
+#define SER_DES_H
+
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <cstring>
+#include <cerrno>
+
+#include <type_traits>
+
+struct file {
+    int fd;
+
+    explicit file(const std::string& path)
+        : fd(open(path.c_str() , O_RDWR | O_CREAT , 0666)){}
```

Comment:
Check the return value and throw an exception. Also it is better to get flags and mode as arguments.